### ISSUE TITLE
Markdown mode: should empty lines (with spaces) escape block lists?

### ISSUE THREAD

**Issue by kevinushey** [ajaxorg/ace#2595]
For example, in the following two screenshots, the header state is disallowed because the tokenizer interprets the indentation as a continuation of a block:

<img width="161" alt="screen shot 2015-07-13 at 10 43 17 am" src="https://cloud.githubusercontent.com/assets/1976582/8656368/01019034-294c-11e5-8452-ba640d0e75b5.png">

<img width="160" alt="screen shot 2015-07-13 at 10 43 22 am" src="https://cloud.githubusercontent.com/assets/1976582/8656369/02659448-294c-11e5-95b1-9efcc9ccd6a3.png">

Should a `list-block` be escaped when there is only whitespace?

https://github.com/ajaxorg/ace/blob/6ebacf96f59182afe7c900b5cee927ba9491668c/lib/ace/mode/markdown_highlight_rules.js#L157-L161

**Comment by nightwing** [comment]
I think it should be escaped when the indentation is equal to the indentation of the starting line.
Currently markdown mode has many similar issues and it's syntax is hard to parse with the tokenizer ace uses.
I wonder if we could use something like https://github.com/markdown-it/markdown-it to make markdown highlighting better
